cloud/endagaweb/views/user.py

```diff
@@ -219,3 +258,94 @@ def update_notify_numbers(request):
+# This view handles the password reset.
+def reset(request):
+    return password_reset(request,
+                          email_template_name=
+                          'dashboard/user_management/reset_email.html',
+                          subject_template_name=
+                          'dashboard/user_management/reset_subject.txt',
+                          post_reset_redirect=reverse('user-management'))
+
+
+# This view handles the changing password to reset.
+def reset_confirm(request, uidb64=None, token=None):
+    return password_reset_confirm(request, uidb64=uidb64,
+                                  template_name=
+                                  'dashboard/user_management/reset_confirm.html',
+                                  token=token, post_reset_redirect=
+                                  reverse('success'))
+
+
+def success(request):
+    return render(request, "dashboard/user_management/success.html")
+
+
+@login_required(login_url='/login/')
+def role_default_permissions(request):
+    if request.method == 'GET':
+        role = request.GET['role']
+        permission_set = ['credit', 'graph', 'report', "smsbroadcast", "tower",
+                          "bts", "subscriber", "network",
+                          "notification", "usageevent"]
+
+        business_analyst = ['view_graph', 'view_report', 'view_bts',
+                            'view_subscriber', 'view_network']
+
+        loader = ['view_graph', 'view_report', 'view_bts', 'view_subscriber',
+                  'view_network', 'change_subscriber', 'change_network',
+                  'add_subscriber', 'add_sms', 'add_credit', 'download_graph']
+
+        partner = ['view_graph', 'view_report', 'view_bts', 'view_subscriber',
+                   'view_network', 'edit_subscriber', 'edit_network',
+                   'add_subscriber', 'add_sms', 'download_graph']
+
+        content_type = ContentType.objects.filter(app_label='endagaweb',
+                                                  model__in=
+                                                  permission_set).values_list('id', flat=True)
+        permission = Permission.objects.filter(
+            content_type__in=content_type).values_list('id', flat=True)
+        role_permission = []
+        if role == 'Business Analyst':
+            role_permission = Permission.objects.filter(
+                codename__in=business_analyst).values_list('id', flat=True)
+        elif role == 'Loader':
+            role_permission = Permission.objects.filter(
+                codename__in=loader).values_list('id', flat=True)
+        elif role == 'Partner':
+            role_permission = Permission.objects.filter(
+                codename__in=partner).values_list('id', flat=True)
+        else:
+            for i in permission:
+                role_permission.append(i)
+
+        return JsonResponse({'permissions': list(role_permission)})
+    return HttpResponseBadRequest()
+
+def validate_password_strength(value):
```

Comment:
Would be good to have some unit tests to validate that this rejects 'weak' passwords, especially since the regex is kinda incomprehensible. It might be clearer if the code was structured as a conjunction of three simpler regex checks: one for a number, one for a letter, and one for a special character.